plutus-core/prelude/PlutusPrelude.hs

```diff
@@ -3,6 +3,7 @@
 {-# LANGUAGE FlexibleInstances     #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RankNTypes            #-}
```

Comment:
where is this pragma needed?

Reply:
cos `Fold` is a type synonym for some terrible lens type